tequilapi/endpoints/service.go

```diff
@@ -56,28 +52,32 @@ type serviceList []serviceInfo
 // swagger:model ServiceInfoDTO
 type serviceInfo struct {
 	// example: 6ba7b810-9dad-11d1-80b4-00c04fd430c8
-	ID       string      `json:"id"`
+	ID string `json:"id"`
+
 	Proposal proposalRes `json:"proposal"`
-	// example: Running
-	Status  string         `json:"status"`
-	Options serviceOptions `json:"options"`
-}
 
-type serviceOptions struct {
-	// example: UDP
-	Protocol string `json:"protocol"`
-	// example: 1190
-	Port int `json:"port"`
+	// example: Running
+	Status string `json:"status"`
 }
 
 // ServiceEndpoint struct represents /service resource and it's sub-resources
 type ServiceEndpoint struct {
 	serviceManager ServiceManager
-	optionsParser  map[string]func(json.RawMessage) (service.Options, error)
+	optionsParser  map[string]ServiceOptionsParser
 }
 
+// ServiceOptionsParser parses request to service specific options
+type ServiceOptionsParser func(json.RawMessage) (service.Options, error)
+
+var (
+	// serviceTypeInvalid represents service type which is unknown to node
+	serviceTypeInvalid = "<invalid>"
```

Comment:
Looks awkward to introduce new special service type just for separating parsing and validation steps - why not just invoke `.toServiceType` inside `.validateServiceRequest` to avoid adding temporary fictional service types?

Reply:
This is pattern that was already fallowed `node`'s codebase and yes it's a bit Goish, but we Gophers at the end